vhost-device-gpio/src/backend.rs

```diff
@@ -196,7 +197,7 @@ fn start_backend(args: GpioArgs) -> Result<()> {
     let mut handles = Vec::new();
 
     for i in 0..config.socket_count {
-        let socket = config.socket_path.to_owned() + &i.to_string();
+        let socket = config.socket_path.join(i.to_string());
```

Comment:
[`Path::join`](https://doc.rust-lang.org/std/path/struct.Path.html#method.join) does not append the suffix to the file name component, but pushes a new component (e.g. "foo.sock0" vs "foo.sock/0")

Reply:
I guess something like `let socket = config.socket_path.with_extension(format!("sock{i}"));` would work better? but that hardcodes the extension though.

Reply:
https://github.com/rust-vmm/vhost-device/blob/1e6667fbba428f3bfc6c3980596dd03e4894b068/vhost-device-i2c/src/main.rs#L239-L256